torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1549,26 +1549,15 @@ bool IsListConstructIntType(const Value* v) {
   return false;
 }
 
-bool AllGraphInputsStatic(const Graph* g) {
-  for (auto n : g->inputs()) {
-    if (TensorTypePtr input_type = n->type()->cast<TensorType>()) {
-      if (input_type->dim()) {
-        auto shape = input_type->symbolic_sizes();
-        if (!ConstantValueMap::HasShape(n->debugName())) {
-          UpdateShapeConstantValueMap(n, shape);
-        }
-      }
-    }
-  }
-  for (auto n : g->inputs()) {
-    // Some inputs can be non-Tensor type, e.g.,
-    // __torch__.torch.classes.quantized.LinearPackedParamsBase
-    // so we only need check Tensor type here.
-    if (n->type()->cast<TensorType>() && !n->isCompleteTensor()) {
-      return false;
-    }
+bool AllGraphInputsStaticWithCaching(const Graph* g) {
```

Comment:
Could you add the rationale from the pr description to here?

Reply:
Done, thanks!

Reply:
@justinchuby  Let me know if the description that I added makes it clear. I just realized I'm not spelling out the perf implications, so I could tweak that if you'd like.

Reply:
I would include the perf implications if possible because I understand that's may reason we make the change. Thanks!

Reply:
Re-worded it a bit. Take a look. Thanks!